docs/docs/core/entity-service.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Entity Service
+slug: /entity-service
+tags:
+  - Strapi Service
+  - Entity Service
+---
+
+# Entity Service
+The Entity Service is a wrapper around strapi.db.query
+This wrapper uses the decorator pattern so plugins and strapi core packages can insert decorators to change the output of the entity service function
+Examples are: plugins/i18n and the core/upload
+
+## Checklist writing decorator
+- You are always required to call service.(function).call() in your decorator function
+- Always return the `opts`/`options` you get from the parent decorator to service into `service.(function).call()`. you are allowed to add your on stuff to it but the parent opts should always be respected.
+- when using `service.(function).call()` ensure the first parameter is always `this` to not break the call chain
+- Your function should always return the output of `service.(function).call()` how ever you are allowed to add stuff to it.
+- when using any wrap functions do not call them in your decorator they are called by the Main Decorator only
```

Comment:
As previously said, I would phrase this as recommendations:
- Call `service.METHOD.call()` in your decorator, so upstream and downstream decorators are also executed.
```js
const decorator = (service) => ({
    async findMany(uid, opts = {}) {
        const result =  service.findMany.call(this, uid, opts);
        // ... update result
		return result;
	}
}
```
- Pass all arguments of the decorated method into `service.METHOD.call()`. Change, add and mutate as necessary,  but take into consideration other decorators might need those arguments too.
- Use `service.(function).call()` to call the downstream decorators.


I would prefer to have a table with good and bad examples too.

<table>
<tr>
 <td>GOOD 🔥
 <td>BAD 👎
<tr>
 <td>
```
 some code
 ```
 <td>
```
 some code
 ```
</table>

Reply:
@Marc-Roig What would be best having a good and bad example for every point directly bellow it?